app/controllers/root_controller.rb

```diff
@@ -109,12 +110,58 @@ def publication
       end
 
       @interaction_details = prepare_interaction_details(@publication, authority_slug, snac)
+    elsif @publication.format == 'local_transaction'
+      @location = fetch_location @postcode
+
+      if @location.nil?
+        @postcode_error = "fullPostcodeNoMapitMatch"
+      elsif @location.try(:code) == 400
+        @postcode_error = "invalidPostcodeFormat"
+      end
+
+      if !@location.nil? && !@location.try(:code)
+        # Valid postcode and matching location
+        snac = appropriate_snac_code_from_location(@publication, @location)
+
+        if snac
+          # Matching local authority and redirect to publication page
+          # with the local authority name. This is the 100% success state.
+          # The redirect below redirects back to this action with the `part`
+          return redirect_to publication_path(slug: params[:slug], part: slug_for_snac_code(snac))
+        else
+          # No matching local authority.
+          # This points the user towards "Find your LA" which is an
+          # England only service
+          @postcode_error = "noLaMatchLinkToFindLa"
+          @location_error = "formats.local_transaction.no_local_authority_html"
+        end
+      elsif params[:part]
```

Comment:
We're missing the:

```ruby
      elsif params[:authority] && params[:authority][:slug].present?
        return redirect_to publication_path(slug: params[:slug], part: CGI.escape(params[:authority][:slug]))
```

part of this conditional that the `licence` format path will run through if no location is found.  Does that only apply to licence types?